src/illuminatio/test_generator.py

```diff
@@ -127,8 +127,8 @@ def generate_negative_cases_for_incoming_cases(self, isolated_hosts, incoming_te
             runtimes[host_string]["findReachingHosts"] = reaching_host_find_time - host_start_time
             if allowed_hosts_with_ports:
                 allowed_hosts, _ = zip(*allowed_hosts_with_ports)
-                ports_per_host = {host: [port for _host, port in allowed_hosts_with_ports if _host == allowed_host]
-                                  for allowed_host in allowed_hosts}
+                ports_per_host = {host: [port for _host, port in allowed_hosts_with_ports if _host == host]
+                                  for host in allowed_hosts}
```

Comment:
Can we add a unit test to catch such errors the next time ?

Reply:
I added three test cases, PTAL